xc/common/utils/prjxray_routing_import.py

```diff
@@ -1061,6 +1081,24 @@ def import_graph_edges(conn, graph, node_mapping):
             if dest_graph_node not in node_mapping:
                 continue
 
+            src_node, src_node_type = node_mapping[src_graph_node]
+            sink_node, sink_node_type = node_mapping[dest_graph_node]
+
+            pin_node_types = [graph2.NodeType.IPIN, graph2.NodeType.OPIN]
+
+            src_node_is_site_pin = src_node_type in pin_node_types
+            sink_node_is_site_pin = sink_node_type in pin_node_types
+
+            # It may happen that a same CHAN <-> PIN edge is generated and this is unaccepted
```

Comment:
This is bad, because that means that the underlying rr graph is less connected than the true hardware routing graph.  Also this code is too pessimistic.  VPR supports allowing multiple `CHAN <-> PIN` edges if a different switch is used.  So you need to account for that.

Reply:
I have added an assertion to check whether the switch is the same for the same `CHAN <-> PIN` pair.

This was meant to avoid the error generated [here](https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/37095c742471af2e2c8d90faf10ab13156a47942/vpr/src/route/check_rr_graph.cpp#L126-L138)